x-pack/plugins/fleet/public/components/agent_enrollment_flyout/hooks.tsx

```diff
@@ -64,20 +72,71 @@ export function useIsK8sPolicy(agentPolicy?: AgentPolicy) {
           : 'IS_NOT_KUBERNETES'
       );
     }
+
+    async function checkifCSP() {
+      if (!agentPolicy) {
+        setIsCSP({ status: 'IS_LOADING', cloudformationUrl: undefined });
+        return;
+      }
+
+      const cspPackageObj = agentPolicy?.package_policies?.map((pkgPolicy) =>
+        getCSPPackageType(pkgPolicy)
+      );
+
+      const cloudFormationTemplateUrl = getCloudFormationTemplateUrlFromPackagePolicy(agentPolicy);
+
+      if (cspPackageObj?.includes('IS_CSP_KSPM')) {
+        setIsCSP({ status: 'IS_CSP_KSPM', cloudformationUrl: cloudFormationTemplateUrl });
+      } else if (cspPackageObj?.includes('IS_CSP_CSPM')) {
+        setIsCSP({ status: 'IS_CSP_CSPM', cloudformationUrl: cloudFormationTemplateUrl });
+      } else if (cspPackageObj?.includes('IS_CSP_CNVM')) {
+        setIsCSP({ status: 'IS_CSP_CNVM', cloudformationUrl: cloudFormationTemplateUrl });
+      } else {
+        setIsCSP({ status: 'IS_NOT_CSP', cloudformationUrl: cloudFormationTemplateUrl });
+      }
+    }
```

Comment:
Following my previous comments we would be able to remove `checkifCSP` from the `useEffect` and use `useMemo` like this instead:


and use useMemo this instead:

```js
const cloudSecurityIntegration = useMemo(() => {
	if (!agentPolicy) {
		return undefined;
	}

	const integrationType = getCloudSecurityIntegrationTypeFromPackagePolicy(agentPolicy);
	const cloudFormationUrl = getCloudFormationTemplateUrlFromPackagePolicy(agentPolicy);

	return {
		integrationType,
		cloudFormationUrl
	}
}, [agentPolicy]);

return { 
	isK8s, 
	cloudSecurityIntegration
};
```

Which is a simpler logic to maintain, and whenever `integrationType` is undefined we consider as not being a Cloud Security integration